internals-js/src/federation.ts

```diff
@@ -862,6 +868,10 @@ export function setSchemaAsFed2Subgraph(schema: Schema) {
   completeSubgraphSchema(schema);
 }
 
+// This is the full @link declaration as added by `asFed2SubgraphDocument`. It's here primarily for uses by tests that print and match
+// subgraph schema to avoid having to update 20+ tests every time we use a new directive or the order of import changes ...
+export const FEDERATION2_LINK_WTH_FULL_IMPORTS = '@link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@key", "@requires", "@provides", "@external", "@tag", "@extends", "@shareable", "@inaccessible"])';
```

Comment:
It can, but I actually didn't do it intentionally. The main use of this is for test assertions and I'd hate to have some syntax issue go unnoticed because we made the same programming error in the code tested and the code used to test it. In other words, I think there is value in having this laid out in plain-text at least once so we can eyeball any issue, and don't think it's a big deal to have to have to update that one place every so often, it's having to update 20 places that was getting on my nerves.

But I reckon that this is caution on my part that borders on paranoia, so let me know if you're not convinced and would prefer generating it as I don't feel strongly at all.